app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -156,20 +156,20 @@ class ValidationService(
   def validateATLs[D <: DataOrigin](
     pageModels: List[PageModel[Visibility]],
     formModelVisibilityOptics: FormModelVisibilityOptics[D]
-  ): Future[ValidatedType[Unit]] =
+  ): ValidatedType[Unit] =
     pageModels
       .flatMap(_.allATLRepeatsWhiles)
-      .distinct
-      .traverse { includeIf =>
-        for {
-          result <- booleanExprEval.eval(formModelVisibilityOptics)(includeIf.booleanExpr)
-        } yield
-        // Returns invalid if the condition is met, otherwise valid for `repeatWhile`
-        if (result) {
+      .groupBy { case (key, _) => key }
+      .map(_._2.flatMap(_._2))
```

Comment:
Can we make this a bit more explicit (and reduce `_2`) like this:
```suggestion
     .values
     .map(_.flatMap(_._2))
```